app/forms/nsm/youth_court_fee_form.rb

```diff
@@ -0,0 +1,48 @@
+module Nsm
+  class YouthCourtFeeForm < BaseAdjustmentForm
+    COMMENT_FIELD = 'youth_court_fee_adjustment_comment'.freeze
+    LINKED_CLASS = V1::AdditionalFee
+
+    attribute :remove_youth_court_fee
+    attribute :youth_court_fee_adjustment_comment
+
+    def save
+      return false unless valid?
+
+      remove_bool = ActiveModel::Type::Boolean.new.cast(remove_youth_court_fee)
+
+      process_field(value: !remove_bool, field: 'include_youth_court_fee')
+
+      true
+    end
+
+    private
+
+    def ensure_original_field_value_set(field)
+      if selected_record["#{field}_original"].present?
+        selected_record.delete("#{field}_original")
```

Comment:
I _think_ I see what you're trying to do here - you're trying to make it so that if a caseworker removes the youth court fee and then re-edits the form to add it back in again, we revert the state to its original state without any `_original` stuff, right? But because you're shoehorning that logic into a side effect of a method called `ensure_original_field_value_set` and what you're doing is _removing_ the original field, it makes for pretty hairy reading. 

I would suggest _not_ doing this - the system should be resilient enough to cope with there being a now-redundant "_original" field, because it copes with it for every other adjustment form, and it will spare the devs bunch of confusion here.